File: task-03/c++/Sub-Task 3.cpp
#include <iostream>

int main() {
    int rows;
    std::cout << "Enter the number of rows: ";
    std::cin >> rows;

    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < rows - i - 1; j++) {
            std::cout << " ";
        }
        for (int j = 0; j < i + 1; j++) {
            std::cout << "* ";
        }
        std::cout << std::endl;
    }

    for (int i = 0; i < rows - 1; i++) {
        for (int j = 0; j < i + 1; j++) {
            std::cout << " ";
        }
        for (int j = 0; j < rows - i - 1; j++) {
            std::cout << "* ";
        }
        std::cout << std::endl;
    }

    return 0;
}

File: task-03/haskell/Sub-Task 3.hs
main :: IO ()
main = do
    putStrLn "Enter the number of rows:"
    input <- getLine
    let rows = read input :: Int
    
    mapM_ (\i -> do
        putStr (replicate (rows-i-1) ' ')
        putStrLn (concat (replicate (i+1) "* "))
        ) [0..rows-1]

    mapM_ (\i -> do
        putStr (replicate (i+1) ' ') 
        putStrLn (concat (replicate (rows-i-1) "* ")) 
        ) [0..rows-2]
